### PR TITLE
PMM-11707: format json result without marshaler

### PR DESCRIPTION
**Pull request by michal-kralik**
PMM-11707

Link to the Feature Build: SUBMODULES-3113

### CODE REVIEW

**Comment by BupycHuk** [comment]
can we write unit tests for this output?

File: admin/cli/cli.go [hunk 1]
```diff
@@ -151,12 +151,17 @@ func printResponse(opts *flags.GlobalFlags, res commands.Result, err error) erro
 
 func printSuccessResult(opts *flags.GlobalFlags, res commands.Result) {
 	if opts.JSON {
-		b, jErr := json.Marshal(res)
-		if jErr != nil {
-			logrus.Infof("Result: %#v.", res)
-			logrus.Panicf("Failed to marshal result to JSON.\n%s.\nPlease report this bug.", jErr)
+		switch res.(type) {
```
Comment (artemgavrilov) [inline admin/cli/cli.go:154]: I guess we need some `default` here.

File: admin/cli/cli.go [hunk 2]
```diff
@@ -151,12 +151,19 @@ func printResponse(opts *flags.GlobalFlags, res commands.Result, err error) erro
 
 func printSuccessResult(opts *flags.GlobalFlags, res commands.Result) {
 	if opts.JSON {
-		b, jErr := json.Marshal(res)
-		if jErr != nil {
+		switch res.(type) {
+		case json.Marshaler:
+			b, jErr := json.Marshal(res)
```
Comment (percona-csalguero) [inline admin/cli/cli.go:156]: if `res` is a `json.Marshaler`, insn't there a `res.Marshal` instead of calling `json.Marshal(res)`?

**Comment by michal-kralik** [comment]
@percona-csalguero is right. This is not a correct approach. Closing in favor of #1827

**Pull request closed without merging**